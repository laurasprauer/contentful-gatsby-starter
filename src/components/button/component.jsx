import React from 'react';
import PropTypes from 'prop-types';
import * as styles from './styles.module.scss';

export const Button = ({ size, label, width, children, ...props }) => {
  const classes = `
		${styles.container}
		${styles[`${size}`]}
		${styles[`${width}`]}
	`;

  return (
    <button type="button" className={classes} {...props}>
      {label ? label : children}
    </button>
  );
};

Button.propTypes = {
  size: PropTypes.oneOf(['small', 'medium']),
  label: PropTypes.string.isRequired,
  width: PropTypes.oneOf(['auto', 'full']),
  children: PropTypes.node,
};

Button.defaultProps = {
  size: 'medium',
  label: '',
  width: 'auto',
};
