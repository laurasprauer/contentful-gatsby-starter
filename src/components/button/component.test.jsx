import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from './component';

describe('Button', () => {
  it('Default button renders correctly', () => {
    const tree = renderer.create(<Button>Button Text</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
