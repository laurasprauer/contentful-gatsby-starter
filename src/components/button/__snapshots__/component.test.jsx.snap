// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Default button renders correctly 1`] = `
<button
  className="
		undefined
		undefined
		undefined
	"
  type="button"
>
  Button Text
</button>
`;
